x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java

```diff
@@ -804,10 +806,10 @@ public Literal visitTimeEscapedLiteral(TimeEscapedLiteralContext ctx) {
         Source source = source(ctx);
 
         // parse HH:mm:ss
-        DateTime dt = null;
+        LocalTime lt = null;
         try {
-            dt = ISODateTimeFormat.hourMinuteSecond().parseDateTime(string);
-        } catch (IllegalArgumentException ex) {
+            lt = LocalTime.parse(string, ISO_LOCAL_TIME);
```

Comment:
I see ISO_LOCAL_TIME deals, also, with format as `HH:mm` (not only `HH:mm:ss`) and the previous joda implementation seems to have looked at `HH:mm:ss` only... not sure if this is an issue or not, meaning if there are 0 seconds, does it return `12:13` only or `12:13:00`?

Reply:
Will dive into that when we implement the `TIME` data type, since now this method always throws an exception ```throw new SqlIllegalArgumentException("Time (only) literals are not supported; a date component is required as well");```